src/utils/media-utils.js

```diff
@@ -284,9 +284,13 @@ export function injectCustomShaderChunks(obj) {
 
       const newMaterial = material.clone();
       // This will not run if the object is never rendered unbatched, since its unbatched shader will never be compiled
-      newMaterial.onBeforeCompile = shader => {
+      newMaterial.onBeforeCompile = (shader, renderer) => {
         if (!vertexRegex.test(shader.vertexShader)) return;
 
+        if (material.onBeforeCompile) {
+          material.onBeforeCompile(shader, renderer);
```

Comment:
Ah good, was wondering how the double shader patching was going to work, this is pretty clean.